robottelo/host_helpers/satellite_mixins.py

```diff
@@ -107,6 +110,42 @@ def md5_by_url(self, url):
             f'wget -qO - {url} | tee {filename} | md5sum | awk \'{{print $1}}\''
         ).stdout
 
+    def upload_manifest(self, org_id, manifest=None, interface='API', timeout=None):
+        """Upload a manifest with locking, using the requested interface.
+
+        :type org_id: int
+        :type manifest: robottelo.manifests.Manifest
+        :type interface: str
+        :type timeout: int
+
+        :returns: the upload result
+
+        Note: The manifest uploading is strictly locked only when using this
+            function
+
+        """
+
+        if manifest is None:
+            manifest = clone()
+        if timeout is None:
+            # Set the timeout to 1500 seconds to align with the API timeout.
+            # And as we are in locked state, other functions/tests can try to upload the manifest in
+            # other processes and we do not want to be interrupted by the default configuration
+            # ssh_client timeout.
+            timeout = 1500000
+        if interface == 'CLI':
+            with manifest:
+                ssh.get_client().put(manifest, manifest.filename)
+            result = Subscription.upload(
+                {'file': manifest.filename, 'organization-id': org_id}, timeout=timeout
+            )
```

Comment:
### Please test it before changing your code as per my below suggestion

I think we should not need manifest context manager here as we are not operating under manifest context.
```suggestion
            self.put(manifest, manifest.filename)
            result = self.cli.Subscription.upload(
                {'file': manifest.filename, 'organization-id': org_id}, timeout=timeout
            )
```

Reply:
This works. Thanks.